Balancer-Bots/Large-Delegation/src/agent.spec.ts

```diff
@@ -0,0 +1,292 @@
+    severity: FindingSeverity.Info,
+    metadata: {
+      delegator,
+      delegate,
+      amount: amount,
+      supplyPercentage: supplyPercentage,
+    },
+  });
+}
+
+const DEFAULT_CONFIG: AgentConfig = {
+  delegateRegistryAddress: DELEGATE_REGISTRY_ADDRESS,
+  veBalTokenAddress: VE_BAL_TOKEN_ADDRESS,
+};
+
+const ADDRESSES = new Array(10).fill(0).map((el, idx) => createChecksumAddress(`0x${idx + 1}`));
+
+describe("Balancer Large Delegation Bot Test Suite", () => {
+  let provider: ethers.providers.Provider;
+  let mockProvider: MockEthersProvider;
+  let handleBlock: HandleBlock;
+
+  const setBalanceOf = (block: number, tokenAddress: string, account: string, balance: ethers.BigNumberish) => {
+    mockProvider.addCallTo(tokenAddress, block, VE_BAL_TOKEN_IFACE, "balanceOf", {
+      inputs: [account],
+      outputs: [ethers.BigNumber.from(balance)],
+    });
+  };
+
+  const setTotalSupply = (block: number, tokenAddress: string, totalSupply: ethers.BigNumberish) => {
+    mockProvider.addCallTo(tokenAddress, block, VE_BAL_TOKEN_IFACE, "totalSupply", {
+      inputs: [],
+      outputs: [ethers.BigNumber.from(totalSupply)],
+    });
+  };
+
+  beforeEach(() => {
+    SmartCaller.clearCache();
+
+    mockProvider = new MockEthersProvider();
+    provider = mockProvider as unknown as ethers.providers.Provider;
+
+    handleBlock = provideHandleBlock(DEFAULT_CONFIG, provider);
+  });
+
+  it("should return empty findings with an empty logs list", async () => {
+    const blockEvent = new TestBlockEvent();
+
+    expect(await handleBlock(blockEvent)).toStrictEqual([]);
+  });
+
+  it("should ignore target events emitted from another contract", async () => {
+    const blockEvent = new TestBlockEvent().setNumber(0);
+
+    mockProvider.addLogs([getSetDelegateLog(ADDRESSES[0], 0, ADDRESSES[1], "balancer.eth", ADDRESSES[2])]);
+
+    expect(await handleBlock(blockEvent)).toStrictEqual([]);
+    expect(mockProvider.call).not.toHaveBeenCalled();
+  });
+
+  it("should ignore other events emitted from target contract", async () => {
+    const blockEvent = new TestBlockEvent().setNumber(0);
+
+    mockProvider.addLogs([getIrrelevantEvent(DELEGATE_REGISTRY_ADDRESS, 0)]);
+
+    expect(await handleBlock(blockEvent)).toStrictEqual([]);
+    expect(mockProvider.call).not.toHaveBeenCalled();
+  });
+
+  it("should ignore target events emitted in other blocks", async () => {
+    const blockEvent = new TestBlockEvent().setNumber(0);
+
+    mockProvider.addLogs([getSetDelegateLog(DELEGATE_REGISTRY_ADDRESS, 1, ADDRESSES[0], "balancer.eth", ADDRESSES[1])]);
+
+    expect(await handleBlock(blockEvent)).toStrictEqual([]);
+    expect(mockProvider.call).not.toHaveBeenCalled();
+  });
+
+  it("should ignore delegations whose id is not 'balancer.eth'", async () => {
+    const blockEvent = new TestBlockEvent().setNumber(0);
+
+    mockProvider.addLogs([getSetDelegateLog(DELEGATE_REGISTRY_ADDRESS, 1, ADDRESSES[0], "some.id", ADDRESSES[1])]);
```

Comment:
```suggestion
    mockProvider.addLogs([getSetDelegateLog(DELEGATE_REGISTRY_ADDRESS, 0, ADDRESSES[0], "some.id", ADDRESSES[1])]);
```
very small thing. it just makes more sense to give the same block number as it tests the effect of different ids

Reply:
Yup, good catch.